Add Erlang/OTP v. Ubuntu compatibility list

paulo-ferraz-oliveira opened the pull request:
Closes #7.

While this addition does not solve the SSL issue it nonetheless closes the reference, since the compatibility issue is out-of-scope for the GitHub Actions (albeit documenting it).

@ericmj, should a new issue be opened next to the project creating the Erlang/OTP builds for `ubuntu-18.04`?

ericmj commented on README.md:
```diff
@@ -34,6 +34,23 @@ For pre-release Elixir versions, such as `1.11.0-rc.0`, use the full version
+| ubuntu-16.04 | 17 - 23    | ✅
+| ubuntu-18.04 | 17         | 🚫<sup>[1]</sup>
+| ubuntu-18.04 | 18         | 🚫<sup>[1]</sup>
+| ubuntu-18.04 | 19         | 🚫<sup>[1]</sup>
```
I think it would be better to not list them.

paulo-ferraz-oliveira replied:
You mean have `17-19` only?

paulo-ferraz-oliveira replied:
I'm going ahead and pushing this, which I believe is the intention behind your comment.

ericmj replied:
No, I think we should list what is compatible not what is not compatible.

ericmj commented on README.md:
```diff
@@ -34,6 +34,23 @@ For pre-release Elixir versions, such as `1.11.0-rc.0`, use the full version
+| ubuntu-18.04 | 18         | 🚫<sup>[1]</sup>
+| ubuntu-18.04 | 19         | 🚫<sup>[1]</sup>
+| ubuntu-18.04 | 20 - 23    | ✅
+| ubuntu-20.04 | 17 - 23    | ✅
```
```suggestion
| ubuntu-20.04 | 20 - 23    | ✅
```

ericmj commented:
> should a new issue be opened next to the project creating the Erlang/OTP builds for ubuntu-18.04?

Yes.

paulo-ferraz-oliveira commented on README.md:
```diff
@@ -34,6 +34,23 @@ For pre-release Elixir versions, such as `1.11.0-rc.0`, use the full version
+| Ubuntu       | Erlang/OTP | Status
+|-             |-           |-
+| ubuntu-16.04 | 17 - 23    | ✅
+| ubuntu-18.04 | 17         | 🚫<sup>[1]</sup>
+| ubuntu-18.04 | 18         | 🚫<sup>[1]</sup>
+| ubuntu-18.04 | 19         | 🚫<sup>[1]</sup>
```
```suggestion
| ubuntu-18.04 | 17 - 19    | 🚫<sup>[1]</sup>
```

paulo-ferraz-oliveira replied:
@ericmj, like this ^?

paulo-ferraz-oliveira commented:
> > should a new issue be opened next to the project creating the Erlang/OTP builds for ubuntu-18.04?
> 
> Yes.

https://github.com/hexpm/bob/issues/82

paulo-ferraz-oliveira commented:
I squashed, rebased and force-pushed. I also fixed the issue with having HTML in Markdown (not a real issue, per se, but not warranting an exception for this simple case either).

paulo-ferraz-oliveira commented on README.md:
```diff
@@ -34,6 +34,21 @@ For pre-release Elixir versions, such as `1.11.0-rc.0`, use the full version
+| Ubuntu       | Erlang/OTP | Status
+|-             |-           |-
+| ubuntu-16.04 | 17 - 23    | ✅
+| ubuntu-18.04 | 17 - 19    | 🚫[1]
```
```suggestion
```

paulo-ferraz-oliveira commented on README.md:
```diff
@@ -34,6 +34,21 @@ For pre-release Elixir versions, such as `1.11.0-rc.0`, use the full version
 
+### Compatibility between Ubuntu and Erlang/OTP
+
+This list presents the known working version combos between Ubuntu
+and Erlang/OTP, as well as the non-working combos.
```
```suggestion
This list presents the known working version combos between Ubuntu
and Erlang/OTP. Non-listed combos are not guaranteed to work.
```

paulo-ferraz-oliveira commented on README.md:
```diff
@@ -34,6 +34,21 @@ For pre-release Elixir versions, such as `1.11.0-rc.0`, use the full version
+| ubuntu-18.04 | 17 - 19    | 🚫[1]
+| ubuntu-18.04 | 20 - 23    | ✅
+| ubuntu-20.04 | 20 - 23    | ✅
+
+[1]: There is a known SSL compatibility issue so you
+should avoid these versions.
```
```suggestion
```

paulo-ferraz-oliveira commented:
Rebased, squashed and force-pushed.
